tests/Backend/CommonPart1/CreateTableTest.php

```diff
@@ -24,6 +24,89 @@ public function setUp()
         $this->_initEmptyTestBuckets();
     }
 
+    public function testSyntheticPrimaryKey()
+    {
+        $testBucketName = $this->getTestBucketName(__METHOD__);
+        $testBucketStage = self::STAGE_IN;
+        $testBucketId = $testBucketStage . '.c-' . $testBucketName;
+        $tableName = 'testSynthPk';
+        $testTableId = $testBucketId . '.' . $tableName;
+
+        $this->dropBucketIfExists($this->_client, $testBucketId);
+        $testBucketId = $this->_client->createBucket($testBucketName, self::STAGE_IN);
+        $this->_client->createTable(
+            $testBucketId,
+            $tableName,
+            new CsvFile(__DIR__ . '/../../_data/languages.csv')
+        );
+
+        // test spk part of detail
+        $table = $this->_client->getTable($testBucketId . '.' . $tableName);
+        $this->assertSame(false, $table['syntheticPrimaryKeyEnabled']);
+
+        try {
+            $tableId = $this->_client->createTable(
+                $testBucketId,
+                'myTable',
+                new CsvFile(__DIR__ . '/../../_data/languages.csv'),
+                ['syntheticPrimaryKeyEnabled' => 1, 'primaryKey' => 'id,name']
+            );
+            $this->fail('Should have thrown');
+        } catch (ClientException $e) {
+            $this->assertSame(
+                'Synthetic primary key cannot be enabled in sync call, use async call to create the table',
+                $e->getMessage()
+            );
+        }
+
+        try {
+            $tableId = $this->_client->createTableAsync(
+                $testBucketId,
+                'myTable',
+                new CsvFile(__DIR__ . '/../../_data/languages.csv'),
+                ['syntheticPrimaryKeyEnabled' => 1]
+            );
+            $this->fail('Should have thrown');
+        } catch (ClientException $e) {
+            $this->assertSame('Cannot set synthetic primary key if primary key is empty', $e->getMessage());
+        }
+
+        try {
+            $tableId = $this->_client->createTableAsync(
+                $testBucketId,
+                'myTable',
+                new CsvFile(__DIR__ . '/../../_data/languages.csv'),
+                ['syntheticPrimaryKeyEnabled' => 1, 'primaryKey' => 'id,name']
+            );
+            $this->fail('Should have thrown');
+        } catch (ClientException $e) {
+            $this->assertSame('Synthetic primary key is not supported outside Snowflake and ABS', $e->getMessage());
```

Comment:
Tady to vlastně nebude fungovat když to pustíš v SNFLK+ABS suitě.

Reply:
Takže to přesunu do skipped části...

Reply:
Tak tím ten problém posuneš na někoho kdo to pak bude řešit nejspíš na budoucího Tomáše :D

Reply:
Successfully converted CMP (current me problem) to FMP (future me problem). 

Třeba to nějak hacknu. Třeba tím, že budu porovnávat jméno projektu.

Reply:
![image](https://user-images.githubusercontent.com/642928/84884110-97b21b80-b091-11ea-910c-32dfd32ea87f.png)

Tak jsem konečně otestoval, že to spadne jak má na ABS.

Reply:
Ale test je skipnutý zatím.